From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 4 Apr 2016 09:50:06 +0200
Subject: [PATCH] {BUG/TODO} rtl8xxxu: Initialize the return value of two
 functions

TODO: function rtl8xxxu_active_to_lps too?

rtl8xxxu_active_to_emu() and rtl8723bu_active_to_emu() use "return
ret;" without initializing "ret" first.  This is harmless as the return
value is not used by their callers but clang complains:

drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c:5311:6: warning:
      variable 'ret' is used uninitialized whenever 'if' condition is false
      [-Wsometimes-uninitialized]
        if (!count) {
            ^~~~~~
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c
index abdff458b80f..ebd8eb93e863 100644
--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu.c
@@ -5231,7 +5231,7 @@ static void rtl8xxxu_set_ampdu_min_space(struct rtl8xxxu_priv *priv, u8 density)
 static int rtl8xxxu_active_to_emu(struct rtl8xxxu_priv *priv)
 {
 	u8 val8;
-	int count, ret;
+	int count, ret = 0;
 
 	/* Start of rtl8723AU_card_enable_flow */
 	/* Act to Cardemu sequence*/
@@ -5281,7 +5281,7 @@ static int rtl8723bu_active_to_emu(struct rtl8xxxu_priv *priv)
 	u8 val8;
 	u16 val16;
 	u32 val32;
-	int count, ret;
+	int count, ret = 0;
 
 	/* Turn off RF */
 	rtl8xxxu_write8(priv, REG_RF_CTRL, 0);
-- 
